reporter/attrmgr.go

```diff
@@ -0,0 +1,62 @@
+package reporter
+
+import (
+	"fmt"
+
+	common "go.opentelemetry.io/proto/otlp/common/v1"
+	profiles "go.opentelemetry.io/proto/otlp/profiles/v1experimental"
+)
+
+// AttrIndex is an index in the `Profile.attribute_table`.
+type AttrIndex = uint64
+
+// AttrTableManager maintains the `Profile.attribute_table` field for the
+// `profile.Profile` that it was created for. Attributes are automatically
+// deduplicated by the manager.
+type AttrTableManager struct {
+	// indices maps compound keys to the indices in the attribute table.
+	indices map[string]AttrIndex
+
+	// profile being built right now.
+	profile *profiles.Profile
+}
+
+func NewAttrTableManager(profile *profiles.Profile) *AttrTableManager {
+	return &AttrTableManager{
+		indices: make(map[string]AttrIndex),
+		profile: profile,
+	}
+}
+
+func (m *AttrTableManager) AddIntAttr(key string, value int64) AttrIndex {
```

Comment:
Use type `attribute.Key` for key, so the string conversions can be removed from the caller side. (In the future we should use semconv keys only.)

Reply:
d0a6a5b